substrate/frame/verify-signature/src/extension.rs

```diff
@@ -0,0 +1,156 @@
+};
+use sp_weights::Weight;
+
+/// Extension that, if enabled, validates a signature type against the payload constructed from the
+/// call and the rest of the transaction extension pipeline. This extension provides the
+/// functionality that traditionally signed transactions had with the implicit signature checking
+/// implemented in [`Checkable`](sp_runtime::traits::Checkable). It is meant to be placed ahead of
+/// any other extensions that do authorization work in the [`TransactionExtension`] pipeline.
+#[derive(Encode, Decode, Clone, Eq, PartialEq, TypeInfo)]
+#[scale_info(skip_type_params(T))]
+pub enum VerifySignature<T>
+where
+	T: Config + Send + Sync,
+{
+	/// The extension will verify the signature and, if successful, authorize a traditionally
+	/// signed transaction.
+	Signed {
+		/// The signature provided by the transaction submitter.
+		signature: T::Signature,
+		/// The account that signed the payload.
+		account: T::AccountId,
+	},
+	/// The extension is disabled and will be passthrough.
+	Disabled,
+}
+
+impl<T> core::fmt::Debug for VerifySignature<T>
+where
+	T: Config + Send + Sync,
+{
+	#[cfg(feature = "std")]
+	fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
+		write!(f, "VerifySignature")
+	}
+
+	#[cfg(not(feature = "std"))]
+	fn fmt(&self, _: &mut core::fmt::Formatter) -> core::fmt::Result {
+		Ok(())
+	}
+}
+
+impl<T> VerifySignature<T>
+where
+	T: Config + Send + Sync,
+{
+	/// Create a new extension instance that will validate the provided signature.
+	pub fn new_with_signature(signature: T::Signature, account: T::AccountId) -> Self {
+		Self::Signed { signature, account }
+	}
+
+	/// Create a new passthrough extension instance.
+	pub fn new_disabled() -> Self {
+		Self::Disabled
+	}
+}
+
+impl<T> TransactionExtension<T::RuntimeCall> for VerifySignature<T>
+where
+	T: Config + Send + Sync,
+	<T::RuntimeCall as Dispatchable>::RuntimeOrigin: AsTransactionAuthorizedOrigin,
+{
+	const IDENTIFIER: &'static str = "VerifyMultiSignature";
+	type Implicit = ();
+	type Val = ();
+	type Pre = ();
+
+	fn weight(&self, _call: &T::RuntimeCall) -> Weight {
+		match &self {
+			// The benchmarked weight of the payload construction and signature checking.
+			Self::Signed { .. } => T::WeightInfo::verify_signature(),
+			// When the extension is passthrough, it consumes no weight.
+			Self::Disabled => Weight::zero(),
+		}
+	}
+
+	fn validate(
+		&self,
+		mut origin: <T::RuntimeCall as Dispatchable>::RuntimeOrigin,
+		_call: &T::RuntimeCall,
+		_info: &DispatchInfoOf<T::RuntimeCall>,
+		_len: usize,
+		_: (),
+		inherited_implication: &impl Encode,
+	) -> Result<
+		(ValidTransaction, Self::Val, <T::RuntimeCall as Dispatchable>::RuntimeOrigin),
+		TransactionValidityError,
+	> {
+		// If the extension is disabled, return early.
+		let Self::Signed { signature, account } = &self else {
+			return Ok((Default::default(), (), origin))
```

Comment:
A little rust opinion: I find an explicit match more explicit here, this is equivalent to 

```
match &self {
    Self::Signed(_) => {}
    _ => {}
}
```

which is not as good as

```
match &self {
    Self::Signed(_) => {}
    Self::Disabled => {}
}
```

Reply:
Done.